### ISSUE TITLE
[2019 ICML Oral] Do ImageNet Classifiers Generalize to ImageNet?

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#79]
해당 논문은 CIFAR-10과 ImageNet 데이터셋에 대해 evaluation dataset인 CIFAR 10.1과 ImageNet v2를 제안함

또한 CIFAR 10.1과 ImageNet v2는 CIFAR, ImageNet과 똑같은 프로세스를 통해 만들어졌음에도 불구하고 accuracy drop 현상이 관측되며, 해당 현상이 일어난 이유는 distribution gap 때문임을 보임

마지막으로 CIFAR-10, ImageNet으로 학습한 모델은 generalization이 부족함을 보여줌
(쉬운 이미지에 대해서는 잘 맞추는데 어려운 이미지에 대해서 잘 맞추지 못함)

## 1. Dataset Creation Methodology

저자들은 CIFAR-10, ImageNet dataset curation process를 반복하여 최대한 original distribution에 가깝게 test set을 구축
(ImageNet 데이터셋의 경우 validation set을 test set으로 봄)

### 1.1. Gathering Data

Dataset마다 specific bias가 있기에, original distribution을 따르기 위해 동일한 dataset source에서 test set 생성

CIFAR-10의 경우 Tiny Image dataset이 source이며, ImageNet은 Flickr image hosting service이 source임

CIFAR-10과 ImageNet은 모두 2000년도에 생성된 데이터셋이기에 car, cell phone class의 경우 많은 변화가 있기에 similar time frame에 upload된 이미지만 이용

두 데이터셋 모두 class 내에서의 structure가 존재하는데, 이를 위해 각 데이터셋의 documentation을 이용하여 sub-class의 distribution을 최대한 매치했다고 함

### 1.2. Cleaning Data

Tiny Images나 Flickr을 통해 얻은 이미지들은 query와 매우 약하거나 관련이 없는 경우가 많음

라벨이 정확한 high-quality dataset을 구축하기 위해 candidate pool에서 적합한 이미지들을 선택하는 과정이 필수적이며, 이를 위해 dataset annotation을 진행함

기존의 두 데이터셋의 경우, CIFAR-10의 경우 학생들이, ImageNet의 경우 Amazon Mechanical Turk (MTurk)를 이용하여 annotation을 하였음

저자들은 이와 비슷하게 CIFAR-10.1 데이터셋을 2명의 대학원생들이 annotation하며, ImageNet v2의 경우 MTurk를 이용하여 annotation함

두 데이터셋 모두 original labeling instruction에 따라 annotation 과정을 진행하였음

Correctly labeled images를 모은 이후, final test set을 위해 sampling을 진행함

CIFAR-10.1의 경우 2,000장의 이미지를, ImageNet v2의 경우 10,000장의 이미지를 가지도록 sampling
(수가 적다고 생각할 수 있는데, 이는 충분히 커서 괜찮음)

### 2.1. Potential Causes of Accuracy Drops
<img width="616" alt="image" src="https://user-images.githubusercontent.com/49643709/179405059-d659e8ac-141f-4a69-9ec3-7f552a88f562.png">

```
D : original data distribution
S : original test set from data distribution D
D' : new data distribution
S' : new test set from data distribution D'
```

#### 2.1.1. Generalization Gap

New test set S'과 data distribution S'의 차이를 generalization gap이라고 정의하며, 이는 random sampling error 때문에 발생

#### 2.1.2. Adaptivity Gap

Original test set S와 data distribution S의 차이를 adaptivity gap이라고 정의

Generalization gap으로 보지 않는 이유는 대부분의 딥러닝 모델들이 test set에서 성능이 좋게끔 hyperparameter를 tuning하기 때문
(validation에서 성능이 잘 나오는 모델보다는 test set에서 성능이 잘 나오는 모델을 사용하기에)

#### 2.1.3. Distribution Gap

Original data distribution D와 new data distribution D'의 차이를 distribution gap이라고 정의

이는 random effect에 영향을 받지 않고 distribution의 systematic difference에 영향을 받음

### 2.2. Conjecture about Results

```
CIFAR-10

- Small dataset size
    - likely experienced more hyperparameter tuning
        - more adaptive overfitting
    - limited size makes models more susceptible to small changes in the distribution
- Constrained visual environment
    - unable to learn robust representation
```

```
ImageNet

- Large dataset size
    - 24x more training images than CIFAR-10
    - 100x more pixels per image than CIFAR-10
    - ImageNet models generalize more reliably
```

저자들이 두 데이터셋을 분석하여 추측한 바는 위와 같음

실제 실험 결과는 추측과 달리 CIFAR-10 모델들은 adaptive overfitting 현상이 관측되지 않으며 ImageNet 모델들은 더 robust하다고 보이지 않음

### 2.3. A Significant Drop in Accuracy
<img width="915" alt="image" src="https://user-images.githubusercontent.com/49643709/179399680-2f051a44-a977-48dd-9b43-72310c5d1721.png">
<img width="907" alt="image" src="https://user-images.githubusercontent.com/49643709/179399729-495b8dbf-b2f3-4fc2-a5cf-56d941d1ab46.png">

저자들은 다양한 모델들에 대해 실험을 진행하였으며, 결과는 `Table 1`, `Figure 1`과 같음
(모델들의 경우 pre-trained model을 사용하거나 official repository의 training command를 통해 학습하였다고 함)

모든 모델들이 original test set에 비해 new test set에서 large accuracy drop 현상이 일어남

### 2.4. Few Changes in Relative Order
<img width="900" alt="image" src="https://user-images.githubusercontent.com/49643709/179400435-c9243652-91e9-4529-b418-0c6e3b0d7183.png">

`Figure 1`을 보면 알 수 있겠지만 original test set에서 성능이 좋으면 new test set에서도 성능이 좋으리라 예측할 수 있으며, original test set과 new test set의 관계에 대한 linear function은 위와 같음

ImageNet의 slope는 1.1로 CIFAR-10의 slope인 1.7보다 낮음

즉 이는 예상했던 시나리오와 달리, adaptive overfitting이 일어나지 않았음을 보여줌
(adaptivity gap으로 인해 발생한 문제가 아니라는 말)

그렇다면 distribution gap으로 인해 성능 drop이 일어난 것인데, 저자들은 human annotation process에 small variation 때문에 발생한 것이라고 함

## 3. Understanding the Impact of Data Cleaning on ImageNet

저자들은 왜 이러한 현상이 일어나는지를 더 확인하기 위해 ImageNet의 3가지 버전의 test set을 만듬

### 3.1. MTurk Tasks
<img width="908" alt="image" src="https://user-images.githubusercontent.com/49643709/179405761-6824e008-55be-45f2-a3e2-3a3dd4b6f7e0.png">
<img width="666" alt="image" src="https://user-images.githubusercontent.com/49643709/179405780-a83bc18b-245f-4a40-8901-c11159910d02.png">

기존존의 ImageNet UI와 비슷하게 MTurk task는 target class가 주어졌을 때 48개의 candidate images grid가 주어지게 됨

Task description은 original ImageNet instruction을 따왔으며, target class의 definition은 Wikipedia page link를 포함했음

저자들은 MTurk workers에게 'occlusion, other objects, clutter or text in the scene'에 상관 없이 target class가 이미지에 있으면 선택하게되 drawings와 paintings는 피하도록 요청하였음
(ImageNet과 같이)

Quality control을 위해 ImageNet validation set에 있는 이미지를 최소 6장을 random select하여 MTurk task에 포함하였음
(3 from the same class, 3 from a class that is nearby in the WordNet hierarchy)

각 target class마다 최소 10명 이상의 MTurk workers를 채용하여 annotation을 진행함

위 과정을 통해 저자들은 각 class마다 최소 20장의 annotated validation images를 얻게 됨

해당 과정의 key point는 original test set과 new test set에 대해 이미지가 선택된 빈도인 `selection frequency` 알 수 있다는 것임
(what fraction of MTurk workers selected the image in a task for its target class)

### 3.2. Sampling Strategies

```
- Matched Frequency : average selection frequency 0.73
    - estimate the selection frequency distribution for each class from annotated original validation images
    - sample 10 images from our candidate pool for each class according to these class-specific distributions
- Threshold 0.7 : average selection frequency 0.85
    - sample 10 images with selection frequency at least 0.7
- Top Images : average selection frequency 0.93
    - choose 10 images with highest selection frequency
```

Sampling strategy에 따라 3가지 버전의 test set을 생성

Labeling error를 줄이기 위해, 저자들이 직접 이미지를 검수하여 incorrect images를 제거하였음

### 3.3. Results
<img width="880" alt="image" src="https://user-images.githubusercontent.com/49643709/179406453-46a564c4-8cd8-458a-9ff9-ac2a472483fb.png">
<img width="879" alt="image" src="https://user-images.githubusercontent.com/49643709/179407099-b5eb045d-b6c6-4fa5-adb4-77a61faeee3f.png">

3가지 버전의 test set을 실험한 결과는 위와 같으며, 모든 test set에 대해 good linear fit을 보임

### 4.1. Distribution Gap

Original test set acc와 new test set acc가 linear한 관계를 보이며 slope가 1보다 크기 때문에 adaptivity gap이라고 볼 수 없음

즉 accuracy drop 현상이 발생한 원인은 distribution gap 때문임

Original dataset creation process를 최대한 따르려고 노력하였으나, distribution gap으로 인해 accuracy drop 현상이 일어났음

이로부터 현존하는 image classification dataset distribution을 replicate하는 것이 매우 어려움을 알 수 있음

Distribution shift가 일어나는 가장 큰 원인은 다양한 변수들에 영향을 받는 human annotation 때문이라고 볼 수 있음
(quality of human labels such as annotator population, exact task format, compensation, ...)

또한 ImageNet의 경우 많은 class들이 명확한 definition이 없기에, 어떤 원인으로 distribution shift가 일어났는지 확인하기 어려움

그리고 `selection frequency`에 따라 accuracy score가 변화하는 것을 통해, 현존하는 ImageNet model들은 `easy` image는 잘 맞추지만 `hard` image를 맞추는 데에 여전히 어려움을 겪고 있음을 알 수 있음
(current ImageNet models still have difficulty generalizing from easy to hard images)

### 4.2. A Model for the Linear Fit
<img width="850" alt="image" src="https://user-images.githubusercontent.com/49643709/179417199-c4cf8ea7-bd1c-4ca8-99cb-cfe7a21dd148.png">
<img width="850" alt="image" src="https://user-images.githubusercontent.com/49643709/179417625-2f99ff74-5baf-4e0b-a0e2-26545481891b.png">
<img width="850" alt="image" src="https://user-images.githubusercontent.com/49643709/179417258-b7146f0a-1e51-4928-88fc-f4f2d26e0926.png">

Original test set acc와 new test set acc간의 linear relationship을 확인할 수 있으나, probit scaling을 통해 더욱 fit하게 만들 수 있음

Accuracy number는 probability이기에 probit scale을 이용하는 것이 더 자연스러움

### 5.1. Adversarial Examples

CIFAR-10에서 $l_{\infty}$-robustness를 이용하여 학습한 모델을 new test set으로 성능을 측정해도 accuracy drop이 여전히 발생

해당 실험을 통해 adversarial robustness는 distribution shift 현상에 큰 도움이 되지 않지만, 다른 adversarial robustness는 실험을 하지 않았기에 확답할 수 없음

### 5.2. Measuring Human Accuracy
<img width="880" alt="image" src="https://user-images.githubusercontent.com/49643709/179417811-24890788-6fcc-49ed-b3c7-49bb22ecb143.png">

그렇다면 new test set은 사람에게 어려운지에 대해 실험을 진행

사람은 original test set과 new test set 사이의 distribution shift에 대해 영향 받지 않음

즉 이미지는 모델에게만 어렵지 사람에게 어렵지 않음을 보임

**Comment by Jasonlee1995** [comment]
Appendix에 꽤 주목할만한 부분이 있어서 추가

논문에서 distribution gap 때문에 차이가 발생했다고 표현하고 있는데 실질적으로는 그렇게 보이지는 않음
(사람이 봤을 때에는 충분히 구분할 수 있는 hard 이미지를 못맞추는 것 뿐인 느낌?)


## 1. CIFAR-10
### 1.1. Training on Part of Our New Test Set

New test set distribution이 original CIFAR-10 distribution과 다르다면, 학습에 new test set을 추가하면 성능이 올라갈 것

New test set에서 random하게 class-balanced split을 통해 1,000장의 이미지를 학습에 추가하여 실험하며 두번 진행

Data를 추가하지 않고 성능이 84.9%가 나오는데 추가하게 되면 85.1%, 85.4%가 나옴

큰 성능 gain이 없기에, new test set distribution과 original CIFAR-10 distribution의 차이는 크지 않다고 볼 수 있음


### 1.2. Training a Discriminator for Original vs. New Test Set
<img width="1000" alt="image" src="https://user-images.githubusercontent.com/49643709/179674817-41503d46-aac6-4889-92ba-8cb0b7e20c27.png">

Original test set과 new test set을 구분할 수 있는지에 대해 binary classification 실험 진행
(ResNet-32, ResNet-110)

실험 결과 50% 대의 성능을 보이며 original test set과 new test set을 구분하지 못하는 것을 알 수 있음


### 1.3. Hard Images
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/49643709/179675627-6955ea9c-2158-41a0-b2b1-447fcccecd51.png">

모든 모델이 틀린 image에 대해 일부를 보여주며, label은 정확함을 보임
(사람은 잘 구분하나 모델은 틀리더라...)


## 2. ImageNet
